components/remote_settings/src/client.rs

```diff
@@ -48,14 +59,59 @@ impl Client {
             bucket_name,
             collection_name: config.collection_name,
             remote_state: Default::default(),
+            cache,
         })
     }
 
     /// Fetches all records for a collection that can be found in the server,
     /// bucket, and collection defined by the [ClientConfig] used to generate
     /// this [Client].
     pub fn get_records(&self) -> Result<RemoteSettingsResponse> {
-        self.get_records_with_options(&GetItemsOptions::new())
+        match self.cache.as_deref() {
+            None => self.get_records_with_options(&GetItemsOptions::new()),
+            Some(cache) => self.get_records_with_cache(cache),
+        }
+    }
+
+    /// Fetches records, using a cached response for efficiency.
+    fn get_records_with_cache(
+        &self,
+        cache: &dyn RemoteSettingsCache,
+    ) -> Result<RemoteSettingsResponse> {
+        let cached_response = cache.get();
+        let cached_last_modified = cached_response
+            .as_ref()
+            .map(|r| r.last_modified)
+            .unwrap_or(0);
+
+        let response = match cached_response {
```

Comment:
I would also do a `if let` here, but maybe it's just my way of finding it easier to read in this case.

```rust
    let response = if let Some(cached) = cached_response {
        let new = self.get_records_since(cached.last_modified)?;
        cache::merge_cache_and_response(cached, new)
    } else {
        self.get_records()?
    };
```